routers/api/v1/repo/issue.go

```diff
@@ -265,6 +272,15 @@ func EditIssue(ctx *context.APIContext, form api.EditIssueOption) {
 		issue.Content = *form.Body
 	}
 
+	if form.Deadline != nil && !form.Deadline.IsZero() {
+		issue.DeadlineUnix = util.TimeStamp(form.Deadline.Unix())
+	} else {
+		issue.DeadlineUnix = util.TimeStamp(0)
+	}
+	if err := models.UpdateIssueDeadline(issue, ctx.User); err != nil {
+		ctx.Error(500, "UpdateIssueDeadline", err)
```

Comment:
Missing `return`?

Reply:
Probably. I've added one (same for pull.go).